crates/edr_napi/src/ts/solidity_tests.ts

```diff
@@ -0,0 +1,54 @@
+import { StandardTestKind, FuzzTestKind, InvariantTestKind } from "../../index";
+
+export enum SortOrder {
+  Ascending,
+  Descending,
+}
+
+export interface GasUsageFilter {
+  minThreshold?: bigint;
+  maxThreshold?: bigint;
+}
+
+function isStandardTestKind(object: any): object is StandardTestKind {
+  return "consumedGas" in object;
+}
+
+function isFuzzTestKind(object: any): object is FuzzTestKind {
+  return "runs" in object && "meanGas" in object && "medianGas" in object;
+}
+
+export function extractGasUsage(
+  testResults: Array<{
+    name: string;
+    kind: StandardTestKind | FuzzTestKind | InvariantTestKind;
+  }>,
+  filter?: GasUsageFilter,
+  ordering?: SortOrder
+): Array<{ name: string; gas: bigint }> {
+  const gasUsage: Array<{ name: string; gas: bigint }> = [];
+
+  for (const result of testResults) {
+    // Default to zero gas for invariant tests
+    const gas = isStandardTestKind(result.kind)
+      ? result.kind.consumedGas
+      : isFuzzTestKind(result.kind)
```

Comment:
I'd suggest this, which is simpler and more type-safe (the helpers defined above are not typechecked):

```suggestion
    const gas = "consumedGas" in result.kind
      ? result.kind.consumedGas
      : "medianGas" in result.kind
```